alerter.c

```diff
@@ -1,32 +1,66 @@
 #include <stdio.h>
 #include <assert.h>
+#include "alerter.h"
 
 int alertFailureCount = 0;
 
 int networkAlertStub(float celcius) {
-    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+    if( celcius < MAX_CELCIUS )
+    {
+        printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+        // Return 200 for ok
+        // Return 500 for not-ok
+        // stub always succeeds and returns 200
+        return 200;
+    }
+    else
+    {
+	printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+        return 500;
+    }
+}
+
+float convertFarenhitToCelcius(float farenheit)
+{
+   return ((farenheit - 32) * 5 / 9);
 }
 
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
-    if (returnCode != 200) {
+void alertInCelcius(float farenheit, int (*NetworkAlertPtr)(float)) 
+{
+    float celcius  = convertFarenhitToCelcius(farenheit);
+    int returnCode = NetworkAlertPtr(celcius);
+    if (returnCode != 200) 
+    {
         // non-ok response is not an error! Issues happen in life!
         // let us keep a count of failures to report
         // However, this code doesn't count failures!
         // Add a test below to catch this bug. Alter the stub above, if needed.
         alertFailureCount += 0;
     }
-}
+    else
+    {
+        alertFailureCount += 1;
+    }
 
-int main() {
-    alertInCelcius(400.5);
-    alertInCelcius(303.6);
+}
+void checkalertInCelcius(float farenheit)
+{
+    alertInCelcius(farenheit, &networkAlertStub);
     printf("%d alerts failed.\n", alertFailureCount);
+}
+
+void commonAlert(void)
+{
+    checkalertInCelcius(400.5);
+    checkalertInCelcius(303.6);
+    checkalertInCelcius(200);
+    checkalertInCelcius(111.2);
+    assert(alertFailureCount == 1);

```

Comment:
This test passes when any of the above calls increments alertFailureCount. Sharpen the test to assert exactly what you intend.